src/storage/src/hummock/compactor/mod.rs

```diff
@@ -631,3 +643,199 @@ pub fn start_compactor(
+                            acc.insert(table.id, table);
+                            acc
+                        });
+                        let static_filter_key_extractor_manager: Arc<StaticFilterKeyExtractorManager> =
+                            Arc::new(StaticFilterKeyExtractorManager::new(id_to_tables));
+                        let filter_key_extractor_manager =
+                            FilterKeyExtractorManager::StaticFilterKeyExtractorManager(
+                                static_filter_key_extractor_manager,
+                            );
+
+                        let mut output_object_ids_deque: VecDeque<_> = VecDeque::new();
+                        output_object_ids_deque.extend(output_object_ids);
+                        let shared_compactor_object_id_manager =
+                            SharedComapctorObjectIdManager::new(output_object_ids_deque);
+                            match dispatch_task.unwrap() {
+                                dispatch_compaction_task_request::Task::CompactTask(compact_task) => {
+                                    context.running_task_count.fetch_add(1, Ordering::SeqCst);
+                                    let (tx, rx) = tokio::sync::oneshot::channel();
+                                    let task_id = compact_task.task_id;
+                                    shutdown.lock().unwrap().insert(task_id, tx);
+
+                                    let (compact_task, table_stats) = compactor_runner::compact(
+                                        context.clone(),
+                                        compact_task,
+                                        rx,
+                                        Box::new(shared_compactor_object_id_manager),
+                                        filter_key_extractor_manager.clone(),
+                                    )
+                                    .await;
+                                    shutdown.lock().unwrap().remove(&task_id);
+                                    context.running_task_count.fetch_sub(1, Ordering::SeqCst);
+                                    let report_compaction_task_request = ReportCompactionTaskRequest {
+                                        event: Some(ReportCompactionTaskEvent::ReportTask(ReportSharedTask {
+                                            compact_task: Some(compact_task),
+                                            table_stats_change: to_prost_table_stats_map(table_stats),
+                                        })),
+                                    };
+
+                                    match report_client
+                                        .report_compaction_task(report_compaction_task_request)
+                                        .await
+                                    {
+                                        Ok(_) => {}
+                                        Err(e) => tracing::warn!("Failed to report task {task_id:?} . {e:?}"),
```

Comment:
TODO: Consider analyzing this err to decide if we should retry or cancel the task

Reply:
Will add retry strategy in next pr.